x-pack/legacy/plugins/maps/public/shared/layers/tile_layer.js

```diff
@@ -106,4 +106,9 @@ export class TileLayer extends AbstractLayer {
   isLayerLoading() {
     return false;
   }
+
+  isDataLoaded() {
+    // data loaded directly by mapbox
```

Comment:
a `TileLayer` does have data-requests. It uses them to retrieve the URL-template asynchronously (e.g. from EMS).

So here too, it should check if the sourceDataRequest has completed. (similar to `HeatmapLayer`)